### PR TITLE
feat: add generate_local_modules_build_files flag to yarn_install and npm_install rules

### PR DESCRIPTION
**Pull request by mistic**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
The current expected behaviour was the one introduced by https://github.com/bazelbuild/rules_nodejs/pull/2330
However there was a regression and after reanalysing it I believe we will be better served by hide that feature behind a flag other than try to proceed with it in an auto discoverable logic.

## What is the new behavior?
With that new behaviour if you set `generate_local_modules_build_files=False` in both `yarn_install` or `npm_install` rules, BUILD files for `symlinked node_modules` will be also symlinked instead of auto generated.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by mistic** [comment]
@alexeagle do you think we can get this fix in for the next minor?

**Comment by alexeagle** [comment]
It depends if it's merged before then 😊
It seems kinda hacky so maybe will need some discussion during review

**Comment by mistic** [comment]
👍 I've kept the previous feature logic, just removed the current check to automatically determine if the feature should run (as it is not currently working as expected) and just used a variable to control it from the rule instead

**Comment by mistic** [comment]
Solved the conflicts with the upstream

**Review by mattem — APPROVED** [review]

File: internal/npm_install/npm_install.bzl [hunk 1]
```diff
@@ -44,6 +44,29 @@ repository so all files that the package manager depends on must be listed.
...
         default = {},
     ),
+    "generate_local_modules_build_files": attr.bool(
+        default = True,
+        doc = """Enables the BUILD files auto generation for local modules installed with `file:` (npm) or `link:` (yarn)
+
+When using a monorepo it's common to have modules that we want to use locally and
+publish to an external package repository. This can be achieved using a `js_library` rule
+with a `package_name` attribute defined inside the local package `BUILD` file. However,
+if the project relies on the local package dependency with `file:` (npm) or `link:` (yarn) to be used outside Bazel, this
```
Comment (ljharb) [inline internal/npm_install/npm_install.bzl:54]: npm also supports `link:` and yarn also supports `file:`, so i'm not sure why these parentheticals are here?

**Pull request merged**